pkg/kubelet/kubelet_pods.go

```diff
@@ -1562,7 +1562,8 @@ func (kl *Kubelet) GetKubeletContainerLogs(ctx context.Context, podFullName, con
 // getPhase returns the phase of a pod given its container info.
 func getPhase(pod *v1.Pod, info []v1.ContainerStatus, podIsTerminal bool) v1.PodPhase {
 	spec := pod.Spec
-	pendingInitialization := 0
+	pendingRestartableInitContainers := 0
```

Comment:
as implemented it is hard to reason whether the new behavior can be disabled by turning the feature gate off. Can you please make logic more compartmentalized so we can reason about it easier from that perspective.

Reply:
@SergeyKanzhelev I only encapsulated the restartable init containers. If other parts also need to do, I would be happy to do so. If you have any better suggestions, I appreciate your feedback. Thanks.